app/controllers/cloud_object_store_container_controller.rb

```diff
@@ -32,8 +32,98 @@ def self.display_methods
     %w(cloud_object_store_objects)
   end
 
+  def new
+    assert_privileges("cloud_object_store_container_new")
+    @in_a_form = true
+    @storage_manager_choices = {}
+    ExtManagementSystem.all.each do |ems|
+      if ems.supports?(:cloud_object_store_container_create)
+        @storage_manager_choices[ems.name] = ems.id
+      end
+    end
+    @provider_regions = retrieve_provider_regions
+    drop_breadcrumb(
+      :name => _("Add New %{model}") % {:model => ui_lookup(:table => 'cloud_object_store_container')},
+      :url  => "/cloud_object_store_container/new"
+    )
+  end
+
+  def create
+    assert_privileges("cloud_object_store_container_new")
+
+    options = form_params_create
+    ext_management_system = options.delete(:ems)
+
+    # Queue task
+    task_id = CloudObjectStoreContainer.cloud_object_store_container_create_queue(
+      session[:userid],
+      ext_management_system,
+      options
+    )
+
+    if task_id.kind_of?(Integer)
+      initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
+    else
+      add_flash(_("Cloud Object Store Container creation failed: Task start failed"), :error)
+      javascript_flash(:spinner_off => true)
+    end
+  end
+
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    container_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Cloud Object Store Container \"%{name}\" created") % {
+        :name => container_name
+      })
+    else
+      add_flash(_("Unable to create Cloud Object Store Container \"%{name}\": %{details}") % {
+        :name    => container_name,
+        :details => task.message
+      }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show_list"
+  end
+
+  def form_params_create
+    options = {}
+    options[:name] = params[:name] if params[:name]
+
+    # Depending on the storage manager type, collect required form params.
+    case params[:emstype]
+    when "ManageIQ::Providers::Amazon::StorageManager::S3"
+      options[:bucket] = params[:name] if params[:name]
```

Comment:
Understood; the only concern was that name and bucket are duplicated. Name is used for flash messages, bucket for s3 provider. That's why I thought it may be worth considering that the name is checked in the backend if bucket is not provided.: so, if bucket is in options in the backend, use it, otherwise check if there is name and set it to bucket.